src/ngamsServer/ngamsServer/ngamsFileUtils.py

```diff
@@ -73,6 +73,33 @@
 checksum_info = collections.namedtuple('crc_info', 'init method final from_bytes equals')
 
 
+def parse_host_id(host_id):
+    """
+    Parses an NGAS server host ID containing '<hostname>:<port>'
+
+    Parameters:
+
+    host_id:        NGAS server host ID
+
+    Returns:
+
+    host:           NGAS server host name (or IP address)
+
+    domain:         NGAS server domain name (or IP address)
+
+    port:           NGAS server port number
+    """
+    try:
+        host = host_id.split(":")[0]
+        domain = None
+        if "." in host:
+            domain = host.split(".", 1)[-1]
+        port = int(host_id.split(":")[-1])
+        return host, domain, port
+    except IndexError:
+        return None, None, None
```

Comment:
`string.split` always produces a list with at least one element, so indexing with `0` and `-1` will always work, and no `IndexError` should occur.

Reply:
OK. I have removed the exception handling